src/components/documentManager.ts

```diff
@@ -0,0 +1,532 @@
+import { BaseItemDto } from '../api/generated/models/base-item-dto';
+
+import { JellyfinApi } from './jellyfinApi';
+
+import { parseISO8601Date } from '../helpers';
+
+// Duration between each backdrop switch in ms
+const backdropPeriodMs: number = 30000;
+
+export abstract class DocumentManager {
+    // Timer state - so that we don't start the interval more than necessary
+    private static backdropTimer: number = 0;
+
+    // TODO: Not sure if these need to be stored.
+    //displayName: string | null;
+    //status: string;
+    //genres: string | null;
+    //overview: string | null;
+    //playedPercentage: number | null;
+
+    // TODO: Static constructor if that's needed.
+    // For now, this is not in use and won't be run.
+
+    // tool for showItem
+    private static getPrimaryImageUrl(item: BaseItemDto): string | null {
+        if (item.AlbumPrimaryImageTag)
+            return JellyfinApi.createUrl(
+                'Items/' +
```

Comment:
If you didnt know, there's a relatively new feature in JS you can use to make this (imho) more legible than string concatenation - "Template Literals" - so you can ditch the string concat and the return statement would become

```
return JellyfinApi.createUrl(`Items/${item.AlbumId}/Images/Primary?tag=${item.AlbumPrimaryImageTag}`);
```